src/utils/test_context.rs

```diff
@@ -531,110 +559,123 @@ impl<'a> TestContextQuery<'a> {
-                .ccv_channel_ids
-                .get(&(src.clone(), dest.clone()))
-                .cloned()
-        } else {
-            None
-        }
+    fn get_admin_addr(&self) -> Option<&str> {
+        let src = self.src_chain.as_deref()?;
+
+        Some(self.context.chains.get(src)?.admin_addr.as_ref())
     }
 
-    fn get_ibc_denom(self) -> Option<String> {
-        if let (Some(ref src), Some(ref dest)) = (self.src_chain, self.dest_chain) {
-            self.context
-                .ibc_denoms
-                .get(&(src.clone(), dest.clone()))
-                .cloned()
-        } else {
-            None
-        }
+    fn get_native_denom(&self) -> Option<&str> {
+        let src = self.src_chain.as_deref()?;
+
+        Some(self.context.chains.get(src)?.native_denom.as_ref())
     }
 
-    fn get_admin_addr(self) -> Option<String> {
-        if let Some(ref src) = self.src_chain {
-            self.context
-                .chains
-                .get(src)
-                .map(|chain| chain.admin_addr.clone())
-        } else {
-            None
-        }
+    fn get_chain_prefix(&self) -> Option<&str> {
+        let src = self.src_chain.as_deref()?;
+
+        Some(self.context.chains.get(src)?.chain_prefix.as_ref())
     }
 
-    fn get_native_denom(self) -> Option<String> {
-        if let Some(ref src) = self.src_chain {
-            self.context
-                .chains
-                .get(src)
-                .map(|chain| chain.native_denom.clone())
-        } else {
-            None
-        }
+    fn get_code_info(&self) -> Option<Value> {
+        let contract = self
+            .context
+            .get_contract(self.contract_name.as_ref()?)
+            .ok()?;
+        let code_id = contract.code_id?;
+        let chain = self
+            .context
+            .chains
+            .get(self.src_chain.as_deref().unwrap_or(NEUTRON_CHAIN_NAME))?;
+
+        // This will produce a { ... text: "{ 'data_hash': xyz }" }. Get the code info enclosed
+        let resp = chain.rb.query(&format!("q wasm code-info {code_id}"), true);
+
+        let str_info_object = resp["text"].as_str()?;
+        serde_json::from_str(str_info_object).ok()
     }
 
-    fn get_chain_prefix(self) -> Option<String> {
-        if let Some(ref src) = self.src_chain {
-            self.context
-                .chains
-                .get(src)
-                .map(|chain| chain.chain_prefix.clone())
-        } else {
-            None
+    fn get_built_contract_address(&self) -> Option<String> {
+        let code_info = self.get_code_info()?;
+        let code_id_hash = code_info["data_hash"].as_str()?;
+
+        let creator_address = NEUTRON_CHAIN_ADMIN_ADDR;
+        let salt = self.salt_hex_encoded.as_deref()?;
+
+        let chain = self
+            .context
+            .chains
+            .get(self.src_chain.as_deref().unwrap_or(NEUTRON_CHAIN_NAME))?;
+
+        // text field contains built address
```

Comment:
can we move this comment to the line with the match

Reply:
Done!